source/Ship.cpp

```diff
@@ -2620,7 +2633,168 @@ bool Ship::IsDisabled() const
+		{
+			if(weapon->Ammo() && !OutfitCount(weapon->Ammo()))
+				continue;
+			return true;
+		}
+	}
+	return false;
+}
+
+
+
+// Check for enemies in the in the current system if ship is owned by player.
+bool Ship::IsEnemyInEscortSystem() const
+{
+	if(!IsYours())
+		return false;
+	const Government *gov = (CanBeCarried() && GetParent()) ? GetParent()->GetGovernment() : GetGovernment();
+	const std::vector<std::weak_ptr<Ship>> allEscorts = (CanBeCarried() && GetParent()) ? GetParent()->GetEscorts() : GetEscorts();
+	for(const weak_ptr<Ship> &ptr : allEscorts)
+	{
+		shared_ptr<Ship> escort = ptr.lock();
+		if(!escort || escort->IsParked() || escort->IsDestroyed() || GetSystem() != escort->GetSystem() || (!escort->IsYours() && !escort->GetPersonality().IsEscort()))
+			continue;
+		shared_ptr<const Ship> escortTarget = escort->GetTargetShip();
+		if(escortTarget)
+		{
+			if(gov->IsEnemy(escortTarget->GetGovernment()))
+				return true;
+		}
+	}
+	return false;
+}
+
+
+
+// If a ship is nearly out of battery energy and is slow to charge or no charge,
+// then consider it low on energy.
+bool Ship::IsEnergyLow() const
+{
+	double frames = 60.;
+	double idleEnergy = frames * GetIdleEnergyPerFrame();
+	double maxEnergy = Attributes().Get("energy capacity");
+	double totalConsumption = frames * GetEnergyConsumptionPerFrame();
+	double currentEnergy = (CanBeCarried() && GetSystem() && !landingPlanet) ? GetCurrentEnergy() : maxEnergy * .75;
+	//double currentEnergy = GetCurrentEnergy();
+	if(totalConsumption < 0.)
+	{
+		double secondsToEmpty = currentEnergy / (-totalConsumption);
+		// how quickly can it charge under no energy load
+		double secondsToFullCharge = (idleEnergy > 0.) ? (maxEnergy - currentEnergy) / idleEnergy : minimumOperatingTime + 1.;
+		if((secondsToEmpty < minimumOperatingTime && secondsToFullCharge > minimumOperatingTime) || (secondsToEmpty < minimumOperatingTime && idleEnergy <= 0.))
+			return true;
+	}
+	return false;
+}
+
+
+
+// Check if escort fleet is full of fuel.  In this case, the fighters can help
+// replenish the Carrier fuel tanker.
+bool Ship::IsEscortsFullOfFuel() const
+{
+	if(!IsYours())
+		return false;
+	const std::vector<std::weak_ptr<Ship>> allEscorts = (CanBeCarried() && GetParent()) ? GetParent()->GetEscorts() : GetEscorts();
+	for(const weak_ptr<Ship> &ptr : allEscorts)
+	{
+		shared_ptr<Ship> escort = ptr.lock();
+		if(!escort || escort->IsParked() || escort->IsDestroyed() || GetSystem() != escort->GetSystem() || (!escort->IsYours() && !escort->GetPersonality().IsEscort()))
+			continue;
+		// Skip fighters and drones.
+		if(escort->CanBeCarried())
+			continue;
+		if(escort->IsFuelLow())
+			return false;
+	}
+	return true;
+}
+
+
+
+// Out of energy ships will be disabled.  Out of energy fighters will be
+// disabled if far away from the parent.  This is so the parent can board the
+// fighter however the fighter needs to not be disabled in order to board the
+// carrier.
```

Comment:
It's a bit weird that out of energy fighters only get disabled if they're far way. It's not consistent. Why not make it so that the fighter gets "reenabled" as soon as it gets recharged by any ships? (I'm fairly sure this is already the case, but then this part doesn't make sense to me "however the fighter needs to not be disabled in order to board the carrier").

Reply:
> Why not make it so that the fighter gets "reenabled" as soon as it gets recharged by any ships? 

Also in case it wasn't clear it's already the case that fighters can be recharged by any ship.  The carrier in particular picks it up so the ship must be enabled for the "pickup" behavior.

Reply:
This is intended behavior so I am resolving this feedback.